include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1138,6 +1137,16 @@ struct __early_exit_find_or
                 // So we use bool variable state check in the for-loop header.
                 __something_was_found = true;
             }
+
+            // Share found into state between items in our sub-group to early exit if something was found
+            // - this approach is applicable only for __parallel_or_tag (hen we search for any matching data entry)
+            // - for __parallel_find_forward_tag and __parallel_find_backward_tag we should process all data
+            if constexpr (_OrTagType{})
+            {
+                __found_local = __dpl_sycl::__any_of_group(__item_id.get_sub_group(), __found_local);
+                if (__found_local != __init_value)
+                    __something_was_found = true;
```

Comment:
Can we just run `__any_of_group` directly on `__something_was_found` and avoid the branching?

Reply:
This would also remove the requirement for passing `__init_value` introduced here.

Reply:
Fixed